HACDOCS-478: Update Permissions and roles upstream content (#211)
redhat-appstudio/docs.appstudio.io: docs/modules/ROOT/pages/getting-started/roles_persmissions.adoc

= Permissions and roles

> As a workspace owner, you can invite collaborators to your workspace by adding their user ID, such as an external SSO ID. After adding a user to a workspace, you assign them a specific role that determines the actions they can take in that workspace.

Reviewer: It's not like you can really "invite" anyone. You are adding a user. The user you are adding can't opt-out. So not a real invitation.

Also, we only support adding usernames. User ID can mean different things.

The roles available for a workspace, from least access to most access, are the following:

* *Contributor*: For users who actively push to your workspace.

> * *Maintainer*: For users who need to manage the workspace without access to sensitive actions like managing security or deleting a workspace.

Reviewer: No contributor or admin can delete a workspace. A workspace can be only deleted by RHTAP admins on request.

Author: @alexeykazakov is "Owner" an RHTAP admin?

Reviewer: No. There is no such a role as "Owner". We might introduce it in the future but we don't have it now.
HAC/UI uses "Owner" but it's a bug. They are working on fixing it and start using "Admin" instead.

> * *Owner*: For users who need full access to the workspace, including sensitive actions like managing security or deleting a workspace.

Reviewer: It's not "Owner". This role is called "Admin". See https://github.com/redhat-appstudio/book/blob/main/ADR/0011-roles-and-permissions.md

Reviewer 2: I just checked in the UI, and even though we call this role "admin" in the backend, the UI does call it "owner". Let's stick with what's in the UI - that's what the user will see.

= Permissions and roles

As a workspace owner, you can add a collaborator to your workspace with their username and assign them a specific role that determines the actions they can take in that workspace.

The roles available for a workspace, from least to most access, are the following:

* *Contributor*: For users who actively push code to your workspace but who don't need to manage workspace configuration.

* *Maintainer*: For users who need to manage the workspace without access to sensitive actions like managing security.

> * *Owner*: For users who need full access to the workspace, including sensitive actions like managing security.

Reviewer 2: See [RHTAPBUGS-958](https://issues.redhat.com/browse/RHTAPBUGS-958) - I know this says "owner" in the UI right now, but it might change to "admin".

Since that is up in the air, maybe we can merge this and then return to update it from owner to admin if that's the change that the UI team eventually decides to make.

Author: hopefully the process works correctly and i/docs gets notified when the ui changes! but yes, it'd be a new jira/pr.

NOTE: By default, the role for newly invited collaborators is *Contributor*, but you can change this.

== Adding a user to your workspace

.Prerequisites

> * The user you want to add must be logged in to {ProductName}.

Reviewer 2: They don't necessarily have to be logged in _right now_ at the _same time_ as you.

```suggestion
* The user you want to add must have a valid login to {ProductName}.
```
